Update to Site-Based Observations

hangtime79 opened the pull request:
Forked to ensure correct results by aligning measures datatypes also add comments, sections, and headers for easier understanding. Now returning metadata and allows for different return languages, units, and no longer do postal codes need to be prefaced with the correct country code. This is now handled in the extension. Also updated so columns with spaces will no longer be an issue.

hangtime79 committed:
Commit with Current Historical Airport R Code

Original Commit of Historical Airport R Code

hangtime79 committed:
Commit Complete Update

Updated to ensure correct results by aligning measures datatypes also
added comments, sections, and headers for easier understanding.

hangtime79 committed:
Bad Version

hangtime79 committed:
Updated for Version 2.0 -

Forked to ensure correct results by aligning measures datatypes also add
comments, sections, and headers for easier understanding. Now returning
metadata and allows for different return languages, units, and no longer
do postal codes need to be prefaced with the correct country code. This
is now handled in the extension. Also updated so columns with spaces
will no longer be an issue.

hangtime79 committed:
Delete LatLong.R

hangtime79 committed:
Delete UnitTest-TWCoHistoricalAirport.str-

hangtime79 committed:
Delete URL Functions.R

Wenpei commented:
@hangtime79 
It's realy great for you PR, and make this extension more robust and user friendly.

One thing is.
Can you add long time(more than 31 days) process, since old extenison support this.

Below code split start end date, and process it month by month. 
```
  for(day in format(seq(sStartDate,sEndDate,30),"%Y%m%d")){
    dayNumber <- as.Date(day,"%Y%m%d")
    sMonthStartDate <- day#format(day,"%Y%m%d")
    if(dayNumber + 30 > sEndDate )
      sMonthEndDate <- format(sEndDate,"%Y%m%d")
    else
      sMonthEndDate <- format(dayNumber + 30,"%Y%m%d")
    sDateRange = paste("&startDate=", 
                       sMonthStartDate, "&endDate=", 
                       sMonthEndDate, sep="")
    sRequestURL <- generateUrl(sLocationUrl,sDateRange)
    print(sRequestURL)
    resultData <- rbind(resultData,retriveDataFromTWC_OneMonth(sRequestURL))
    print(nrow(resultData))
  }
```

another thing is about 
```
packages <- function(x){
  x <- as.character(match.call()[[2]])
  if (!require(x,character.only=TRUE)){
    install.packages(pkgs=x,repos="http://cran.r-project.org")
    require(x,character.only=TRUE)
  }
}
```
this function try to auto install required package. but you didn't use it in extension.
But I am OK if you don't want it, because it may cause a protention issues, when some package updated(install.packages will install latest package). 

Thank you very much.
Yu Wenpei.

hangtime79 committed:
Version 2.11 - Updates from Requests from Initial Pull Request

Added logic so that date spans greater than 30 days will now create
multiple requests (for example if 1/1/2016 to 2/1/2016 is added, it
would create two requests instead of 1). In addition, I would like to
keep the requirement to install packages. I'm using quite a few and if
they are not installed the extension will fail. I believe in this case
if the package is installed already it will default to it and NOT try to
update.

hangtime79 committed:
Revert "Version 2.11 - Updates from Requests from Initial Pull Request"

This reverts commit e14497a510eb54e5f598f9851733a5695eebe08f.

hangtime79 committed:
Version 2.11 - Updates based on Request of Last Pull Request

Added logic so that date spans greater than 30 days will now create
multiple requests (ex. if 1/1/2016 to 2/1/2016 is passed it would create
two requests since it spans over 30+ days instead of 1). In addition, I
would like to keep the requirement to install packagtes. I'm using quite
a few and if they are not installed the extension will fail. I believe
in this case if the packagte is isntalled already it willd efault to it
and NOT try to update the package.

hangtime79 committed:
Add files via upload

hangtime79 committed:
Add files via upload

hangtime79 committed:
Delete Comments.csv

hangtime79 committed:
Add files via upload

hangtime79 committed:
Delete Comments.csv

hangtime79 committed:
Delete Financial_Trans.csv

hangtime79 committed:
Add files via upload

hangtime79 committed:
Add files via upload

hangtime79 committed:
Delete Financial_Trans_2.csv

hangtime79 committed:
Add files via upload

hangtime79 committed:
Delete Financial_Trans_2.csv

hangtime79 committed:
Add files via upload
